deploy/helm/kubecf/assets/scripts/jobs/cf-cli-6-linux/setup_internal_security_group.sh

```diff
@@ -3,60 +3,218 @@
-
-# Define the security group JSON.
+wait_for_endpoint "${cf_api_ca_cert_pem}" "${CF_API}/v2/info"
+
+# Wait for UAA to be accessible.
+echo "Waiting for UAA to be accessible..."
+wait_for_endpoint "${uaa_ca_cert_pem}" "${UAA_URL}/info"
+
+# Construct the security group JSON definition.
+sec_group_name="internal-${POD_NAME}"
 sec_group_json=$(cat <<EOF
-[{
-    "protocol": "tcp",
-    "description": "Allow traffic to ${POD_NAME}",
-    "destination": "${POD_IP}/32",
-    "ports": "${PORTS}"
-}]
+{
+    "name": "${sec_group_name}",
+    "rules": [{
+        "protocol": "tcp",
+        "description": "Allow traffic to ${POD_NAME}",
+        "destination": "${POD_IP}/32",
+        "ports": "${PORTS}"
+    }]
+}
 EOF
 )
 
-# Create the security group if it doens't exist, otherwise update it.
-sec_group_name="internal-${POD_NAME}"
-if ! cf security-group "${sec_group_name}"; then
-    cf create-security-group "${sec_group_name}" <(echo -n "${sec_group_json}")
-    cf bind-staging-security-group "${sec_group_name}"
-    cf bind-running-security-group "${sec_group_name}"
+# get_access_token returns a token to be used on the CF API calls.
+function get_access_token() {
+    scope="scope=cloud_controller.admin"
+    grant_type="grant_type=client_credentials"
+    client_id="client_id=${CF_USERNAME}"
+    client_secret="client_secret=${CF_PASSWORD}"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${uaa_ca_cert_pem}" \
+        --request POST \
+        --data "${scope}&${grant_type}&${client_id}&${client_secret}" \
+        "${UAA_URL}/oauth/token" \
+        | jq -r '.access_token'
+}
+
+# check_security_group checks if the security group already exists.
+function check_security_group() {
+    local access_token
+    access_token="$1"
+    local total_results
+    total_results=$(
+        curl \
+            --silent --show-error --fail \
+            --cacert "${cf_api_ca_cert_pem}" \
+            --header "Authorization: Bearer ${access_token}" \
+            "${CF_API}/v2/security_groups?q=name:${sec_group_name}" \
+            | jq -r '.total_results'
+    )
+    if [[ "${total_results}" == "0" ]]; then return 1; fi
+}
+
+# get_sec_group_id returns the security group ID.
+function get_sec_group_id() {
+    local access_token
+    access_token="$1"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${cf_api_ca_cert_pem}" \
+        --header "Authorization: Bearer ${access_token}" \
+        "${CF_API}/v2/security_groups?q=name:${sec_group_name}" \
+        | jq -r '.resources[0].metadata.guid'
+}
+
+# create_security_group creates the new security group.
+function create_security_group() {
+    local access_token
+    access_token="$1"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${cf_api_ca_cert_pem}" \
+        --header "Authorization: Bearer ${access_token}" \
+        --header "Content-Type: application/json" \
+        --request POST \
+        --data "${sec_group_json}" \
```

Comment:
Should we be passing in `${cf_api_ca_cert_pem}` and `${sec_group_json}` explicitly? Global variables flying around doesn't seem awesome…

(Please _don't_ do this if you think it will involve creating classes in bash.)

Reply:
I was in the fence between passing them explicitly or using the value of the outer scope. I'll pass them explicitly, makes it easier to reason.

Reply:
Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/485/commits/da0e3bd19f83bf58ffaa06ff6972d0ed46930e8e.